### PR TITLE
Remove gzip compatibility layer since we now no longer support Python 2.6

### PR DESCRIPTION
**Pull request by astrofrog**

### CODE REVIEW

**Comment by astrofrog** [comment]
@embray - can you review this since it makes some changes in astropy.io.fits?

**Comment by taldcroft** [comment]
Looks good to me.  I would ask the same question again about making our compat gzip module raise an exception and make sure tests pass.  

One question also about `astropy/tests/tests/test_imports.py`.  In your branch this still has references to the removed modules.  Is that intended?
```
# Compatibility subpackages that should only be used on Python 2
_py2_packages = set([
    'astropy.extern.configobj_py2',
    'astropy.utils.compat._fractions_py2',
    'astropy.utils.compat._gzip_py2',
    'astropy.utils.compat._odict_py2',
    'astropy.utils.compat._subprocess_py2'
])

# Same but for Python 3
_py3_packages = set([
    'astropy.extern.configobj_py3',
    'astropy.utils.compat._gzip_py3'
])
```

**Comment by astrofrog** [comment]
@taldcroft - I did check that no test failures occurred if I raise an exception. I'll clean up the list of modules in ``test_imports`` - good catch (but not critical)

File: CHANGES.rst [hunk 1]
```diff
@@ -164,6 +164,9 @@ Bug fixes
 
 - ``astropy.utils``
 
+- The astropy.utils.compat.gzip module has now been deprecated. Use the
+  Python 'gzip' module directly instead. [#4464]
+
```
Comment (embray) [inline CHANGES.rst]: This needs to be indented 2 spaces.

File: astropy/io/fits/tests/test_core.py [hunk 2]
```diff
@@ -683,7 +683,7 @@ def test_read_open_astropy_gzip_file(self):
         compatibility copy of the ``gzip`` module.
         """
 
-        from ....utils.compat import gzip
+        import gzip
```
Comment (embray) [inline astropy/io/fits/tests/test_core.py]: There's no reason this needs to be an in-line import (there wasn't any reason I can think of previously too, but might as well remove it while we're here).

**Comment by embray** [comment]
Much better....

File: CHANGES.rst (hunk 1 above)
Comment (astrofrog) [inline CHANGES.rst]: Fixed

File: astropy/io/fits/tests/test_core.py (hunk 2 above)
Comment (astrofrog) [inline astropy/io/fits/tests/test_core.py]: Fixed

**Comment by astrofrog** [comment]
@embray - done. Ready to merge? :)

**Pull request merged**